### PR TITLE
fixes https://github.com/uProxy/uproxy/issues/1166

### PR DESCRIPTION
**Pull request by salomegeo**
Tested manually

### CODE REVIEW

**Comment by ryscheng** [comment]
Cool hidden flag, but I'm curious, why are we ever calling sendMessage when status !== ONLINE?

File: src/core/socialprovider.js [hunk 1]
```diff
@@ -302,6 +303,12 @@ XMPPSocialProvider.prototype.sendMessage = function(to, msg, continuation) {
             body += this.messages[to][i].message;
           }
           message.t(body);
+
```
Comment (ryscheng) [inline src/core/socialprovider.js]: This block is only called if status !=== "ONLINE". Why are we sending messages to clients that are either OFFLINE or ONLINE_WITH_OTHER_APP?

**Comment by salomegeo** [comment]
I don't know, currently we only send messages to clients that are ONLINE from uproxy.

**Comment by ryscheng** [comment]
If that's the case, then the code you added should never be called and wouldn't fix the issue right?
Do you have a sense of where the problem of
https://github.com/uProxy/uproxy/issues/1166
is coming from?

I can see a couple possibilities:
- uProxy forgets to check whether a user is ONLINE before sending a message, causing it to potentially send messages to non-uProxy clients.
- the social provider is incorrectly classifying clients as ONLINE when they are not uProxy clients
- non-Gmail friend accounts somehow exhibit different behavior that what we expect.

Do you have a way to replicate the bug in https://github.com/uProxy/uproxy/issues/1166?

**Comment by salomegeo** [comment]
I'm pretty sure I tested this, like it was archiving messages before this change and then it stopped. but I agree it makes no sense. let me try it again

**Comment by salomegeo** [comment]
Oh I see, chat demos are using this path, but uproxy uses the other.

I moved it out of if/else block. PTAL

**Comment by salomegeo** [comment]
On Mon, Mar 30, 2015 at 3:08 PM, Raymond Cheng <notifications@github.com>
wrote:

> If that's the case, then the code you added should never be called and
> wouldn't fix the issue right?
> Do you have a sense of where the problem of
> uProxy/uproxy#1166 <https://github.com/uProxy/uproxy/issues/1166>
> is coming from?
>
> I can see a couple possibilities:
>
>    - uProxy forgets to check whether a user is ONLINE before sending a
>    message, causing it to potentially send messages to non-uProxy clients.
>
>
>    - the social provider is incorrectly classifying clients as ONLINE
>    when they are not uProxy clients
>
> No this is happening when users are not logged in with any other
client(only uProxy), and I examine urpxoy data sturctures and that matches
to what I saw


>
>    - non-Gmail friend accounts somehow exhibit different behavior that
>    what we expect.
>
> This is happening with gmail accounts too.


>
>
> Do you have a way to replicate the bug in uProxy/uproxy#1166
> <https://github.com/uProxy/uproxy/issues/1166>?
>
Yes just log in with two accounts in uproxy and go to "Chats" in gmail.

**Comment by ryscheng** [comment]
Interesting. Shouldn't the nosave option only be set if the user is "ONLINE"?

**Comment by salomegeo** [comment]
I think that makes sense, but we'd have to fix chat demo, but that's low priority I guess

**Comment by ryscheng** [comment]
Can you clarify what about the chat demo would need to be fixed?

**Comment by ryscheng** [comment]
:+1:

**Pull request merged**